components/core/src/clp_s/JsonParser.cpp

```diff
@@ -557,6 +621,303 @@ int32_t JsonParser::add_metadata_field(std::string_view const field_name, NodeTy
     return m_archive_writer->add_node(metadata_subtree_id, type, field_name);
 }
 
+auto JsonParser::get_archive_node_type(
+        clp::ffi::SchemaTree const& tree,
+        std::pair<clp::ffi::SchemaTree::Node::id_t, std::optional<clp::ffi::Value>> const& kv_pair
+) -> NodeType {
+    clp::ffi::SchemaTree::Node const& tree_node = tree.get_node(kv_pair.first);
+    clp::ffi::SchemaTree::Node::Type const ir_node_type = tree_node.get_type();
+    bool const node_has_value = kv_pair.second.has_value();
+    std::optional<clp::ffi::Value> node_value{};
+    if (node_has_value) {
+        node_value = kv_pair.second.value();
```

Comment:
If we're doing this check does node_value still need to be wrapped in std::optional?

Seems like we could rewrite this bit as `auto &node_value = kv_pair.second()`

And replace instances of node_has_value with `node_value.has_value()`.

Either that or declare node_value as `clp::ffi::Value` directly without the optional.